### DISCUSSION TITLE
Synchronize `.devcontainer` between `pretext new` and the `pretext-codespace` repository

### DISCUSSION THREAD

**Discussion by oscarlevin** [PreTeXtBook/pretext-cli#455] [Ideas]
We currently have two main ways that an author can start a pretext project.
1. They can create a repository from the [pretext-codespace](https://github.com/PreTeXtBook/pretext-codespace) repository.
2. They can run `pretext new` locally (or `pretext init --refresh` on a previously created project).

For both of these, the user gets a `.devcontainer` folder that let's them use codespaces.  We should try to keep these containers identical.  But that's tricky because part of creating a project from the template repository should be removing the directions to create the repository after it is made.  Currently that is accomplished by calling `pretext new -d .` as part of that `.devcontainer`.   But that recreates the `.devcontainer` and prompts the user to rebuild.  

Any ideas are very welcome.  @StevenClontz, @bjones1 ?

**Comment by bjones1** [comment]
How about only overwriting the `devcontainer.json`  file if it's different during `pretext new -d`?

**Comment by StevenClontz** [comment]
My pitch: as part of the `deploy-*` actions (besides nightly) we create a fresh `pretext new` document that's pushed to `pretext-codespaces`. So no human ever maintains that repo - we just update it automatically as part of our release process for CLI.

**Comment by oscarlevin** [comment]
This is now implemented, and the docker image is being rebuild on each release as well.

**Comment by bjones1** [comment]
Nice!
